### PR TITLE
[google_maps_flutter] Fix NNBD migration mistake in example

### PR DESCRIPTION
**Pull request by stuartmorgan**
The TileOverlay portion of the example app incorrectly constructed a
Set<TileOverlay?> and then attempted to cast it to a Set<TileOverlay>,
rather than making a set of non-nullable types as required by the API.

Fixes https://github.com/flutter/flutter/issues/79242

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

**Comment by stuartmorgan** [comment]
No test for this change since it looks like this integration test is one of the ones that uses the APIs directly rather than driving the example app.

File: packages/google_maps_flutter/google_maps_flutter/pubspec.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: google_maps_flutter
 description: A Flutter plugin for integrating Google Maps in iOS and Android applications.
 homepage: https://github.com/flutter/plugins/tree/master/packages/google_maps_flutter/google_maps_flutter
-version: 2.0.1
+version: 2.0.2
```
Comment (cyanglaz) [inline packages/google_maps_flutter/google_maps_flutter/pubspec.yaml:4]: Probably not necessary? Updating the version has no harm either.

**Review by cyanglaz — APPROVED** [review]
LGTM

File: packages/google_maps_flutter/google_maps_flutter/pubspec.yaml (hunk 1 above)
Comment (stuartmorgan) [reply packages/google_maps_flutter/google_maps_flutter/pubspec.yaml:4]: I thought more of the example was visible on pub.dev than it is.

I think it's worth mentioning in CHANGELOG, so I'll leave it. (Something I think we should consider is a way to add CHANGELOG lines for cases where it's not worth actually publishing yet. E.g., allow for "<NEXT>" as a version as long as the version in pubspec hasn't changed.)

Comment (ditman) [reply packages/google_maps_flutter/google_maps_flutter/pubspec.yaml:4]: Dart people do `-dev` releases for these cases, until they need to publish for real, and then they collapse everything into a next version with multiple entries.

**Review by ditman — APPROVED** [review]

**Pull request merged**